techmatters/terraso-mobile-client
# feat: form sheet component

## Description

Bring `FormOverlaySheet` in compliance with design spec, simplifying its implementation to remove unsupported use-cases. Specifically:
- Sheet 'done' header is now a reusable named component included in the sheet itself, rather than embedded in `Select` component
- Sheet header stays pinned while content scrolls
- Sheet now snaps to both 50% and 100% of screen
- Sheet now scrolls content with specified amount of padding at bottom

Supporting these changes:
- Move the `NavigationContainer` higher in the `App.tsx` stack to enable proper content scrolling in the sheet component when embedded in a modal.
- Also migrate the `ImagePicker` and site soil modals to use `StandaloneOverlaySheet` and `InfoSheet` respectively per discussion with design
- Extract some common logic to make the three sheet variants more concise in implementation.

### Checklist

### Related Issues
#1774

### Verification steps

Examine instances of overlay sheets and check that they still retain expected behavior. (For `Select` components, `InfoSheet` components including for site depth intervals, and for the camera/image picker for site soil colors.)
